### PR TITLE
Shared Resource Environment on cloud.rc

### LINKED ISSUE #325: Document how to create a shared resource environment on cloud.rc
An openstack project space is simply a context for allocating resource and enforcing quotas.  By default, each cloud.rc user is assigned their own project space with a "starter" quota.  This project has the same name as as their login id.  They can use this project to deploy their own instances.   They are also the only user that is a member of this project.

It may be desirable for multiple people to collaborate in the operation of cloud instances.  There are multiple approaches to accomplish this:  

A simple option that works with the default configuration above, is simply to add the desired co-sysadmin's public ssh key to an instance.  This makes is possible for additional users to participate in instance operations.  Care should be taken to avoid multiple sysadmins from making changes to the instances.  As always, you need to trust your sysadmins.  

In this scenario, instances remain running in the default project space where they launched the instance.   The user is the only person who can grant access to this project.   The cloud.rc web-UI restricts access only to members explicitly added to the project (i.e. the user alone). However, the user can create application credentials to grant access to other applications via the OpenStack API.  Again, this should only done with trusted sysadmins.   Anyone granted control over the project space can delete instances or other resources.  Just like in real life, you should only give room keys to people you trust not to throw your computer in the trash. 

A more sophisticated project configuration is possible by adding multiple users to a project.  This configuration is desirable if there are multiple people sharing operational responsibilities for shared resources, e.g. a lab with multiple instances.  In this case, a project can be created for the lab and multiple users added to the lab.  All the same cautions for the simple solution above apply.  Anyone with access to the shared lab space can delete instances and other resources. Users in this configuration will be able to access the shared project in the cloud.rc web-UI or grant other applications through the creation of application credentials.  Again, it is important that you trust your sysadmins.

It is possible to move from the simple default project configuration with one user to the more sophisticated multi-user project.  This requesting the creation of a named, shared project space and identify the user who should have administrative access to this project.   Users can then move their own cloud instances into this share project space through normal cloud operations.  An outline of this project is:
- save your instance as image
- share the image with the shared project
- accept the image into shared project
- launch an instance of the image in the shared project space
- similar steps are followed for moving data volumes between projects

The cloud.rc documentation needs to be extended to describe these operations.

There are many helpful resources online:
https://www.cac.cornell.edu/wiki/index.php?title=Share_An_Image_In_Openstack
https://docs.ukcloud.com/articles/openstack/ostack-how-move-resources.html

### PR DESCRIPTION
**Pull request by iam4tune**
# Pull Request

First commit to create the page, and include instructions for sharing cloud resources in a project space.

## Proposed Changes

Developed a draft article detailing the ways to share cloud resources in a project space, included instructions on how to share a cloud resource (instance).

## Related Issues

Fixes #325

### CODE REVIEW

File: docs/uab_cloud/project_space_mgmt.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+The created Image can then be launched, following the same instructions for creating an [Instance](../uab_cloud/tutorial/instances.md). This method would be most ideal if you want to recreate an environment for performing an analysis, but would prefer the workflow be run on different VMs, or to separate datasets or create some form of access restriction on particular research.
+
+### Creating an Image From a Volume
+
+There are detailed instructions here on how to create an image from a Volume, [here](snapshots.md#creating-a-volume-snapshot).
+
+### Using a Key Pair to SSH
+
+Another way to access a created instance would be to create a public key and private key for your local machine, and then share this public key with the creator of the instance. As creator of the instance you would need to add individual public keys of persons who you would want to access the created VM into the `authorized_keys` file. You can edit this file by using the command below, add the shared public key in a new line inside the file (copy and paste). Save the file and follow instructions here for remote accessing your instance using [`SSH`](remote_access.md).
```
Comment (wwarriner) [inline docs/uab_cloud/project_space_mgmt.md:75]: See previous note about adding multiple people to Key Pair.

**Review by wwarriner — CHANGES_REQUESTED** [review]
I think it's a great start! I think we could improve the wording, and there is information I have that we will want to add about having multiple people added at once when creating a key pair. We can talk next week about all this.

File: docs/uab_cloud/project_space_mgmt.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+To enjoy the resources available on UAB Cloud, you would have to send in a request to the UAB IT Research Computing Team via email [support@listserv.uab.edu](mailto:support@listserv.uab.edu). In the email, please state clearly your needs and the resources you would require for your lab. Your request should also include members of your lab to be included in the project space, and a preferred name (usually same as lab project folder on Cheaha). Instances created within this project space can be accessed by other members of your lab.
+
+## Creating an Instance in a Shared Project Space
+
+The steps for creating an instance in a project are the same as creating an instance in your personal project space, you can find a detailed guide here in our [Cloud Tutorial](tutorial/index.md). You would however, need to contact us to create a project space for your lab, as well as users who would be members of the shared project space.
+
+All members of a project space can access all instances within the shared project space on UAB Cloud RC. They can do this by navigating through the landing page that doubles as the dashboard, the image below highlights the exact section.
```
[the remark this answers is not in the document]
Comment (iam4tune) [reply docs/uab_cloud/project_space_mgmt.md:31]: I will effect the changes as per your suggestions in preparation for Wednesday.

File: docs/uab_cloud/project_space_mgmt.md (hunk 1 above)
Comment (iam4tune) [reply docs/uab_cloud/project_space_mgmt.md:75]: Noted

File: docs/uab_cloud/project_space_mgmt.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+![!RC Dashboard Screenshot showing Project Spaces](images/rc_move_project.png)
+
+## Creating an Instance in a Shared Project Space
+
+The steps for creating an `Instance` in a project are the same as creating an `Instance` in your personal project space, you can find a detailed guide here in our [Cloud Tutorial](tutorial/index.md). You would however, need to contact us to create a project space for your lab, as well as users who would be members of the shared project space.
+
+All members of a project space can see (and manage) all `Instances` within the shared project space on UAB Cloud.rc. They can do this by navigating through the landing page that doubles as the dashboard, the image below highlights the exact section. While members can see and create images or snapshots of an `Instance`, not all members can access the `Instance`. To do this, `SSH` key pairs would have to be created and added, see section on [Using a Key Pair to SSH](#using-a-key-pair-to-ssh).
```
Comment (wwarriner) [inline docs/uab_cloud/project_space_mgmt.md:31]: `ssh` in lower case, like the command is typed in Linux. Or without backticks in upper case, like the initialism "**S**ecure **SH**ell". I prefer SSH over `ssh`, except in code blocks.

File: docs/uab_cloud/project_space_mgmt.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+![!RC Dashboard Screenshot showing Project Space Name](images/rc_proj_dashboard.png)
+
+When in the project space, select `Compute` and then `Instances` to see available `Instances` in your project space.
```
Comment (wwarriner) [inline docs/uab_cloud/project_space_mgmt.md:35]: Use double quotes for text descriptions of "Menu Items" used in GUIs.

**Review by wwarriner — CHANGES_REQUESTED** [review]
Looks great, please check comments for a few general corrections.

File: mkdocs.yml
Comment (wwarriner) [inline mkdocs.yml:1]: One last thing, let's spell out the filename in full like `project_space_management.md`.

**Review by wwarriner — CHANGES_REQUESTED** [review]

**Review by wwarriner — APPROVED** [review]
Looks great!

**Pull request merged**